Source/WebCore/dom/Node.cpp

```diff
@@ -2102,6 +2106,21 @@ Element* Node::enclosingLinkEventParentOrSelf()
     return nullptr;
 }
 
+void Node::expandAnyDetailsAncestor()
+{
+    if (!document().settings().detailsAutoExpandEnabled())
+        return;
+    ShouldNotFireMutationEventsScope scope(document());
+    for (auto& element : ancestorsOfType<HTMLElement>(*this)) {
+        if (is<HTMLSummaryElement>(element))
+            continue;
+        if (is<HTMLDetailsElement>(element) && !element.hasAttributeWithoutSynchronization(HTMLNames::openAttr)) {
+            downcast<HTMLDetailsElement>(element).toggleOpen();
```

Comment:
We should be holding a Ref / RefPtr to the element before calling toggleOpen() on it.

Reply:
I see that Tim's proposal to switch to dynamicDowncast<> also addresses my concern too. Never mind.